### ISSUE TITLE
Dashboard: "Get certificiate" button is displayed even though user has paid for course

### ISSUE THREAD

**Issue by pdpinch** [mitodl/mitxonline#675]
### Steps to Reproduce

I haven't been able to reproduce this with my own user, but you can see it on RC if you masquerade as the user `ddoyle`. 

Their dashboard indicates that they are enrolled in the free track for the Microeconomics course that starts Jan 12, 2022 -- but they also have a receipt for that course (in fact, they have three!) 

<img width="1139" alt="image" src="https://user-images.githubusercontent.com/430126/177844175-b2e8380a-11d4-459a-8ffc-5aef52d8886f.png">

Receipt 1: https://rc.mitxonline.mit.edu/orders/receipt/172/
Receipt 2: https://rc.mitxonline.mit.edu/orders/receipt/285/
Receipt 3: https://rc.mitxonline.mit.edu/orders/receipt/292/

### Expected Behavior

1. If the user has paid for a course run, the course listing on the dashboard should be labelled `Enrolled in certificate course`
2. If the user has pad for a course run, the "get a certificate" button should not be visible. 
 

### Actual Behavior

- See above

### Related Issues

- Don't allow the user to pay for the same course run twice 

\

**Comment by rhysyngsun** [comment]
This definitely looks like an issue with the learner's state. I'd presume we would have an error in Sentry to indicate what went wrong.

See https://rc.mitxonline.mit.edu/admin/courses/courserunenrollment/281/change/?_changelist_filters=q%3Dddoyle

**Comment by pdpinch** [comment]
here's what I found in sentry:

https://sentry.io/organizations/mit-office-of-digital-learning/issues/3077938630/?environment=rc&project=5864687&query=is%3Aunresolved+ddoyle&statsPeriod=14d

```
EdX API error enrolling user dana-doyle (ddoyle@mit.edu) in course run 'course-v1:MITx+14.750x+3T2021'.
Response - code: 400, content: {"message":"The [verified] course mode is expired or otherwise unavailable for course run [course-v1:MITx+14.750x+3T2021].","course_details":{"course_id":"course-v1:MITx+14.750x+3T2021","course_name":"Political Economy and Economic Development","enrollment_start":"2021-10-10T00:00:00Z","enrollment_end":"2022-12-08T23:59:00Z","course_start":"2022-04-18T15:00:00Z","course_e...
```

Seems like we need some way to handle this better.

**Comment by pdpinch** [comment]
#676 will make this better, but it won't hide the button.

**Comment by arslanashraf7** [comment]
My initial understanding is that this might be a duplicate of https://github.com/mitodl/mitxonline/issues/762. Or vice versa.

**Comment by rhysyngsun** [comment]
@arslanashraf7 yes, I'm pretty sure addressing #762 will fix this, but we'll leave this issue open for now so that we can be sure to circle back and verify it since there were multiple things at play here.

**Comment by arslanashraf7** [comment]
> here's what I found in sentry:
> 
> https://sentry.io/organizations/mit-office-of-digital-learning/issues/3077938630/?environment=rc&project=5864687&query=is%3Aunresolved+ddoyle&statsPeriod=14d
> 
> ```
> EdX API error enrolling user dana-doyle (ddoyle@mit.edu) in course run 'course-v1:MITx+14.750x+3T2021'.
> Response - code: 400, content: {"message":"The [verified] course mode is expired or otherwise unavailable for course run [course-v1:MITx+14.750x+3T2021].","course_details":{"course_id":"course-v1:MITx+14.750x+3T2021","course_name":"Political Economy and Economic Development","enrollment_start":"2021-10-10T00:00:00Z","enrollment_end":"2022-12-08T23:59:00Z","course_start":"2022-04-18T15:00:00Z","course_e...
> ```
> 
> Seems like we need some way to handle this better.

As per this comment, What happened was that the course mode was not usable and this could happen for a couple of reasons (_Though this is not specifically related to upgrading because we might see this in the case of free courses too_):


1- There was no Course Mode present in edX against that course (e.g. `Audit`, `Verified`)
2- The Course Mode was present but the `Upgrade deadline` has passed.

So we have dealt with the Course Mode update/create in edX the user should be able to enroll again OR if we have [IGNORE_EDX_FAILURES](https://github.com/mitodl/mitxonline/blob/main/main/features.py#L7) set to true, the enrollment will still be created in MITxOnline and upon `openedx.tasks.retry_failed_edx_enrollments` this should then enroll the user in edX automatically.

**Comment by pdpinch** [comment]
As far as I know, #762 has been addressed, but I still see the "Get Certificate" button displaying on the dashboard of this user. How can/should I address this?

**Comment by rhysyngsun** [comment]
@pdpinch #762 fixes this issue from occurring in the future, that particular user's issue required a manual repair. I applied these steps:

- Updated that enrollment to be verified and edx_enrolled=False
- Ran `./manage.py retry_edx_enrollment` for that user

I've confirmed they're now showing up as verified in their dashboard.

The orders are still in a weird state, but it should be functional going forward.
